docs/sources/operations/blocking-queries.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Blocking Queries
+weight: 60
+---
+# Blocking Queries
+
+In certain situations, you may not be able to control the queries being sent to your Loki installation. These queries
+may be intentionally or unintentionally expensive to run, and they may affect the overall stability or cost of running
+your service.
+
+You can block queries using [per-tenant overrides](../configuration/#runtime-configuration-file), like so:
+
+```yaml
+overrides:
+  "tenant-id":
+    blocked_queries:
+      # block this query exactly
+      - pattern: 'sum(rate({env="prod"}[1m]))'
+
+      # block any query matching this regex pattern 
+      - pattern: '.*prod.*'
+        regex: true
+
+      # block all metric queries
+      - types: metric
+
+      # block any filter or limited queries matching this regex pattern 
+      - pattern: '.*prod.*'
+        regex: true
+        types: filter,limited
```

Comment:
Is the order of the patterns important? If yes we should mention that first match will be first serve

Reply:
That's a good call; the order is deterministic since this is a list - i'll update the docs.